tests/integration/replication.tcl

```diff
@@ -409,47 +409,112 @@ test {slave fails full sync and diskless load swapdb recovers it} {
             # 10ms per key, with 800 keys is 8 seconds
             $master config set rdb-key-save-delay 10000
 
+            set loglines [count_log_lines -1]
+
             # Start the replication process...
             $slave slaveof $master_host $master_port
 
-            # wait for the slave to start reading the rdb
+            # wait for the replica to start reading the rdb
             wait_for_condition 50 100 {
-                [s -1 loading] eq 1
+                [s -1 async_loading] eq 1
```

Comment:
That's what I was afraid. Could be that some management tool is checking for the "loading" from INFO instead of sending a command (for example a load balancer).
But at the same time, it's not a breaking change to show INFO loading = 1 without responding commands with `-LOADING`. Won't change behavior for those that simply upgrade redis, it's just that it will require tweaking to take advantage of this implementation IF they rely on INFO.
I'd be happy to change it to not have the async_loading flag and set loading to 1 instead during async_loading.